File: cs/libldt3/model/objekte/UntersuchungsergebnisZytologie.cs
/*
 * Copyright 2016-2024  Christoph Brill <opensource@christophbrill.de>
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
using libldt3.attributes;
using libldt3.model;
using libldt3.model.enums;
using libldt3.model.regel.erlaubt;
using libldt3.model.regel.format;
using libldt3.model.regel.kontext;

namespace libldt3
{
    namespace model
    {
        namespace objekte
        {
            /// <summary>
            /// In diesem Objekt können weitere Ergebnisse aus dem Bereich Zytologie
            /// transportiert werden.
            /// </summary>
            [Objekt(Value = "0063", Kontextregeln = new[] { typeof(K053), typeof(K076), typeof(K080), typeof(K081), typeof(K082), typeof(K096), typeof(K100), typeof(K119) })]
            public class UntersuchungsergebnisZytologie : Kontext
            {
                [Objekt]
                public class ErgebnisId : Kontext
                {
                    public string Value;
                    [Feld(Value = "7320", Feldart = Feldart.bedingt_kann)]
                    [Regelsatz(Laenge = 1)]
                    public RecallEmpfohlen_ RecallEmpfohlen;

                }
                [Objekt]
                public class RecallEmpfohlen_ : Kontext
                {
                    public bool? Value;
                    [Feld(Value = "8154", Feldart = Feldart.bedingt_muss)]
                    [Regelsatz(Laenge = 9)]
                    public Timestamp Timestamp;

                }
                [Objekt(Kontextregeln = new[] { typeof(K053) })]
                public class KatalogAnforderbareLeistungenId : Kontext
                {
                    public KatalogIdAnforderbareLeistungen? Value;
                    [Feld(Value = "7352", Feldart = Feldart.bedingt_muss)]
                    [Regelsatz(MaxLaenge = 60)]
                    public string UrlKataloge;
                    [Feld(Value = "7251", Feldart = Feldart.bedingt_kann)]
                    [Regelsatz(MaxLaenge = 60)]
                    public string BezeichnungDesVerwendetenKataloges;
                    [Feld(Value = "7365", Feldart = Feldart.bedingt_muss)]
                    [Regelsatz(MaxLaenge = 20)]
                    public AnalysenId AnalysenId;

                }
                [Objekt]
                public class AnalysenId : Kontext
                {
                    public string Value;
                    [Feld(Value = "7366", Feldart = Feldart.bedingt_kann)]
                    [Regelsatz(MaxLaenge = 60)]
                    public string LangbezeichnungAngefordertenLeistung;

                }
                [Objekt]
                public class TestIdent : Kontext
                {
                    public string Value;
                    [Feld(Value = "8411", Feldart = Feldart.bedingt_muss)]
                    [Regelsatz(MaxLaenge = 60)]
                    public string Testbezeichnung;

                }
                [Objekt]
                public class Ergebnisstatus : Kontext
                {
                    public TestStatus? Value;
                    [Feld(Value = "8422", Feldart = Feldart.bedingt_muss)]
                    [Regelsatz(MaxLaenge = 2)]
                    public IList<GrenzwertindikatorLaborwerte> GrenzwertindikatorLaborwerte;

                }
                [Objekt(Kontextregeln = new[] { typeof(K099) })]
                public class GrenzwertindikatorLaborwerte : Kontext
                {
                    public Grenzwertindikator? Value;
                    [Feld(Value = "8126", Feldart = Feldart.bedingt_muss)]
                    [Regelsatz(Laenge = 28)]
                    public FehlermeldungAufmerksamkeit FehlermeldungAufmerksamkeit;

                }
                [Objekt]
                public class HpvBefundObjUntersuchungsergebnisZytologie : Kontext
                {
                    public bool? Value;
                    [Feld(Value = "7402", Feldart = Feldart.bedingt_kann)]
                    [Regelsatz(Value = new[] { typeof(F024) }, MaxLaenge = 120)]
                    public IList<HighRiskTyp> HighRiskTyp;
                    [Feld(Value = "7404", Feldart = Feldart.bedingt_kann)]
                    [Regelsatz(Value = new[] { typeof(F024) }, MaxLaenge = 120)]
                    public IList<LowRiskTyp> LowRiskTyp;

                }
                [Objekt]
                public class HighRiskTyp : Kontext
                {
                    public string Value;
                    [Feld(Value = "7401", Feldart = Feldart.bedingt_muss)]
                    [Regelsatz(Laenge = 1)]
                    public HpvHrTestergebnis? HighRisk;

                }
                [Objekt]
                public class LowRiskTyp : Kontext
                {
                    public string Value;
                    [Feld(Value = "7403", Feldart = Feldart.bedingt_muss)]
                    [Regelsatz(Laenge = 1)]
                    public HpvHrTestergebnis? LowRisk;

                }
                [Objekt]
                public class Gruppe : Kontext
                {
                    public string Value;
                    [Feld(Value = "7413", Feldart = Feldart.bedingt_muss)]
                    [Regelsatz(Value = new[] { typeof(E028) }, MaxLaenge = 4)]
                    public string CodierungGruppe;

                }
                [Feld(Value = "7304", Feldart = Feldart.muss)]
                [Regelsatz(MaxLaenge = 60)]
                public ErgebnisId ErgebnisId;
                [Feld(Value = "7364", Feldart = Feldart.muss)]
                [Regelsatz(MaxLaenge = 60)]
                public IList<string> ProbengefaessIdent;
                [Feld(Value = "7260", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(Laenge = 1)]
                public IList<KatalogAnforderbareLeistungenId> KatalogAnforderbareLeistungenId;
                [Feld(Value = "8410", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(MaxLaenge = 20)]
                public IList<TestIdent> TestIdent;
                [Feld(Value = "8418", Feldart = Feldart.muss)]
                [Regelsatz(Laenge = 2)]
                public Ergebnisstatus Ergebnisstatus;
                [Feld(Value = "8237", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(Laenge = 12)]
                public Fliesstext Ergebnistext;
                [Feld(Value = "7368", Feldart = Feldart.kann)]
                [Regelsatz(Laenge = 1)]
                public bool? ZellmaterialNichtVerwertbar;
                [Feld(Value = "7400", Feldart = Feldart.kann)]
                [Regelsatz(Laenge = 1)]
                public HpvBefundObjUntersuchungsergebnisZytologie HpvBefundObjUntersuchungsergebnisZytologie;
                [Feld(Value = "7414", Feldart = Feldart.kann)]
                [Regelsatz(Value = new[] { typeof(E028) }, MaxLaenge = 5)]
                public Gruppe Gruppe;
                [Feld(Value = "7418", Feldart = Feldart.kann)]
                [Regelsatz(Laenge = 1)]
                public Ergebnis? P16Ki67;
                [Feld(Value = "7419", Feldart = Feldart.kann)]
                [Regelsatz(Laenge = 1)]
                public Ergebnis? L1;
                [Feld(Value = "7422", Feldart = Feldart.kann)]
                [Regelsatz(Laenge = 1)]
                public Ergebnis? Chlamydien;
                [Feld(Value = "7425", Feldart = Feldart.kann)]
                [Regelsatz(Laenge = 1)]
                public Ergebnis2? ExtragynaekologischeZytologie;
                [Feld(Value = "7426", Feldart = Feldart.kann)]
                [Regelsatz(Laenge = 1)]
                public Ergebnis? NeisseriaGonorrhoeae;
                [Feld(Value = "8126", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(Laenge = 28)]
                public FehlermeldungAufmerksamkeit FehlermeldungAufmerksamkeit;
                [Feld(Value = "8220", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(Laenge = 36)]
                public Timestamp TimestampEingangserfassungMaterial;
                [Feld(Value = "8222", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(Laenge = 25)]
                public Timestamp TimestampBeginnAnalytik;
                [Feld(Value = "8223", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(Laenge = 28)]
                public Timestamp TimestampErgebniserstellung;
                [Feld(Value = "8224", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(Laenge = 22)]
                public Timestamp TimestampQmErfassung;
                [Feld(Value = "8225", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(Laenge = 17)]
                public Timestamp TimestampMessung;
                [Feld(Value = "8167", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(Laenge = 26)]
                public IList<Fliesstext> ZusaetzlicheInformationen;
                [Feld(Value = "8110", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(Laenge = 6)]
                public IList<Anhang> AnhangObjUntersuchungsergebnisZytologie;
                [Feld(Value = "8141", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(Laenge = 13)]
                public Namenskennung Namenskennung;
                [Feld(Value = "8158", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(Laenge = 23)]
                public Untersuchungsabrechnung Untersuchungsabrechnung;
                [Feld(Value = "7429", Feldart = Feldart.kann)]
                [Regelsatz(MaxLaenge = 990)]
                public string DrgHinweis;
                [Feld(Value = "3473", Feldart = Feldart.kann)]
                [Regelsatz(Laenge = 1)]
                public bool? UntersuchungsergebnisDurchAuftragslaboratoriumErstellt;

            }
        }
    }
}

File: java/src/main/java/libldt3/model/objekte/UntersuchungsergebnisZytologie.java
/*
 * Copyright 2016-2024  Christoph Brill &lt;opensource@christophbrill.de&gt;
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package libldt3.model.objekte;

import java.util.List;

import libldt3.annotations.Feld;
import libldt3.annotations.Feldart;
import libldt3.annotations.Objekt;
import libldt3.annotations.Regelsatz;
import libldt3.model.Kontext;
import libldt3.model.enums.Ergebnis;
import libldt3.model.enums.Ergebnis2;
import libldt3.model.enums.Grenzwertindikator;
import libldt3.model.enums.HpvHrTestergebnis;
import libldt3.model.enums.KatalogIdAnforderbareLeistungen;
import libldt3.model.enums.TestStatus;
import libldt3.model.regel.erlaubt.E028;
import libldt3.model.regel.format.F024;
import libldt3.model.regel.kontext.K053;
import libldt3.model.regel.kontext.K076;
import libldt3.model.regel.kontext.K080;
import libldt3.model.regel.kontext.K081;
import libldt3.model.regel.kontext.K082;
import libldt3.model.regel.kontext.K096;
import libldt3.model.regel.kontext.K099;
import libldt3.model.regel.kontext.K100;
import libldt3.model.regel.kontext.K119;

/**
 * In diesem Objekt können weitere Ergebnisse aus dem Bereich Zytologie
 * transportiert werden.
 */
@Objekt(value = "0063", kontextregeln = {K053.class, K076.class, K080.class, K081.class, K096.class, K100.class, K119.class})
public class UntersuchungsergebnisZytologie implements Kontext {

    @Objekt
    public static class ErgebnisId implements Kontext {
        @SuppressWarnings("unused")
        public String value;
        @Feld(value = "7320", feldart = Feldart.bedingt_kann)
        @Regelsatz(laenge = 1)
        public RecallEmpfohlen recallEmpfohlen;
    }

    @Objekt
    public static class RecallEmpfohlen implements Kontext {
        @SuppressWarnings("unused")
        public Boolean value;
        @Feld(value = "8154", feldart = Feldart.bedingt_muss)
        @Regelsatz(laenge = 9)
        public Timestamp timestamp;
    }

    @Objekt(kontextregeln = K053.class)
    public static class KatalogAnforderbareLeistungenId implements Kontext {
        @SuppressWarnings("unused")
        public KatalogIdAnforderbareLeistungen value;
        @Feld(value = "7352", feldart = Feldart.bedingt_muss)
        @Regelsatz(maxLaenge = 60)
        public String urlKataloge;
        @Feld(value = "7251", feldart = Feldart.bedingt_kann)
        @Regelsatz(maxLaenge = 60)
        public String bezeichnungDesVerwendetenKataloges;
        @Feld(value = "7365", feldart = Feldart.bedingt_muss)
        @Regelsatz(maxLaenge = 20)
        public AnalysenId analysenId;
    }

    @Objekt
    public static class AnalysenId implements Kontext {
        @SuppressWarnings("unused")
        public String value;
        @Feld(value = "7366", feldart = Feldart.bedingt_kann)
        @Regelsatz(maxLaenge = 60)
        public String langbezeichnungAngefordertenLeistung;
    }

    @Objekt
    public static class TestIdent implements Kontext {
        @SuppressWarnings("unused")
        public String value;
        @Feld(value = "8411", feldart = Feldart.bedingt_muss)
        @Regelsatz(maxLaenge = 60)
        public String testbezeichnung;
    }

    @Objekt
    public static class Ergebnisstatus implements Kontext {
        @SuppressWarnings("unused")
        public TestStatus value;
        @Feld(value = "8422", feldart = Feldart.bedingt_muss)
        @Regelsatz(maxLaenge = 2)
        public List<GrenzwertindikatorLaborwerte> grenzwertindikatorLaborwerte;
    }

    @Objekt(kontextregeln = K099.class)
    public static class GrenzwertindikatorLaborwerte implements Kontext {
        @SuppressWarnings("unused")
        public Grenzwertindikator value;
        @Feld(value = "8126", feldart = Feldart.bedingt_muss)
        @Regelsatz(laenge = 28)
        public FehlermeldungAufmerksamkeit fehlermeldungAufmerksamkeit;
    }

    @Objekt
    public static class HpvBefundObjUntersuchungsergebnisZytologie implements Kontext {
        @SuppressWarnings("unused")
        public Boolean value;
        @Feld(value = "7402", feldart = Feldart.bedingt_kann)
        @Regelsatz(value = F024.class, maxLaenge = 120)
        public List<HighRiskTyp> highRiskTyp;
        @Feld(value = "7404", feldart = Feldart.bedingt_kann)
        @Regelsatz(value = F024.class, maxLaenge = 120)
        public List<LowRiskTyp> lowRiskTyp;
    }

    @Objekt
    public static class HighRiskTyp implements Kontext {
        @SuppressWarnings("unused")
        public String value;
        @Feld(value = "7401", feldart = Feldart.bedingt_muss)
        @Regelsatz(laenge = 1)
        public HpvHrTestergebnis highRisk;
    }

    @Objekt
    public static class LowRiskTyp implements Kontext {
        @SuppressWarnings("unused")
        public String value;
        @Feld(value = "7403", feldart = Feldart.bedingt_muss)
        @Regelsatz(laenge = 1)
        public HpvHrTestergebnis lowRisk;
    }

    @Objekt
    public static class Gruppe implements Kontext {
        @SuppressWarnings("unused")
        public String value;
        @Feld(value = "7413", feldart = Feldart.bedingt_muss)
        @Regelsatz(value = E028.class, maxLaenge = 4)
        public String codierungGruppe;
    }

    @Feld(value = "7304", feldart = Feldart.muss)
    @Regelsatz(maxLaenge = 60)
    public ErgebnisId ergebnisId;
    @Feld(value = "7364", feldart = Feldart.muss)
    @Regelsatz(maxLaenge = 60)
    public List<String> probengefaessIdent;
    @Feld(value = "7260", feldart = Feldart.bedingt_muss)
    @Regelsatz(laenge = 1)
    public List<KatalogAnforderbareLeistungenId> katalogAnforderbareLeistungenId;
    @Feld(value = "8410", feldart = Feldart.bedingt_muss)
    @Regelsatz(maxLaenge = 20)
    public List<TestIdent> testIdent;
    @Feld(value = "8418", feldart = Feldart.muss)
    @Regelsatz(laenge = 2)
    public Ergebnisstatus ergebnisstatus;
    @Feld(value = "8237", feldart = Feldart.bedingt_muss)
    @Regelsatz(laenge = 12)
    public Fliesstext ergebnistext;
    @Feld(value = "7368", feldart = Feldart.kann)
    @Regelsatz(laenge = 1)
    public Boolean zellmaterialNichtVerwertbar;
    @Feld(value = "7400", feldart = Feldart.kann)
    @Regelsatz(laenge = 1)
    public HpvBefundObjUntersuchungsergebnisZytologie hpvBefundObjUntersuchungsergebnisZytologie;
    @Feld(value = "7414", feldart = Feldart.kann)
    @Regelsatz(value = E028.class, maxLaenge = 5)
    public Gruppe gruppe;
    @Feld(value = "7418", feldart = Feldart.kann)
    @Regelsatz(laenge = 1)
    public Ergebnis p16Ki67;
    @Feld(value = "7419", feldart = Feldart.kann)
    @Regelsatz(laenge = 1)
    public Ergebnis l1;
    @Feld(value = "7422", feldart = Feldart.kann)
    @Regelsatz(laenge = 1)
    public Ergebnis chlamydien;
    @Feld(value = "7425", feldart = Feldart.kann)
    @Regelsatz(laenge = 1)
    public Ergebnis2 extragynaekologischeZytologie;
    @Feld(value = "7426", feldart = Feldart.kann)
    @Regelsatz(laenge = 1)
    public Ergebnis neisseriaGonorrhoeae;
    @Feld(value = "8126", feldart = Feldart.bedingt_muss)
    @Regelsatz(laenge = 28)
    public FehlermeldungAufmerksamkeit fehlermeldungAufmerksamkeit;
    @Feld(value = "8220", feldart = Feldart.bedingt_muss)
    @Regelsatz(laenge = 36)
    public Timestamp timestampEingangserfassungMaterial;
    @Feld(value = "8222", feldart = Feldart.bedingt_muss)
    @Regelsatz(laenge = 25)
    public Timestamp timestampBeginnAnalytik;
    @Feld(value = "8223", feldart = Feldart.bedingt_muss)
    @Regelsatz(laenge = 28)
    public Timestamp timestampErgebniserstellung;
    @Feld(value = "8224", feldart = Feldart.bedingt_muss)
    @Regelsatz(laenge = 22)
    public Timestamp timestampQmErfassung;
    @Feld(value = "8225", feldart = Feldart.bedingt_muss)
    @Regelsatz(laenge = 17)
    public Timestamp timestampMessung;
    @Feld(value = "8167", feldart = Feldart.bedingt_muss)
    @Regelsatz(laenge = 26)
    public List<Fliesstext> zusaetzlicheInformationen;
    @Feld(value = "8110", feldart = Feldart.bedingt_muss)
    @Regelsatz(laenge = 6)
    public List<Anhang> anhangObjUntersuchungsergebnisZytologie;
    @Feld(value = "8141", feldart = Feldart.bedingt_muss)
    @Regelsatz(laenge = 13)
    public Namenskennung namenskennung;
    @Feld(value = "8158", feldart = Feldart.bedingt_muss)
    @Regelsatz(laenge = 23)
    public Untersuchungsabrechnung untersuchungsabrechnung;
    @Feld(value = "7429", feldart = Feldart.kann)
    @Regelsatz(maxLaenge = 990)
    public String drgHinweis;
    @Feld(value = "3473", feldart = Feldart.kann)
    @Regelsatz(laenge = 1)
    public Boolean untersuchungsergebnisDurchAuftragslaboratoriumErstellt;

}
